src/server/cmd/job-shim/main.go

```diff
@@ -30,6 +38,74 @@ func main() {
 	env.Main(do, &appEnv{})
 }
 
+func downloadInput(c *client.APIClient, commitMounts []*fuse.CommitMount) error {
+	for _, commitMount := range commitMounts {
+		if commitMount.Alias == "prev" || commitMount.Alias == "out" {
+			continue
+		}
+		repo := commitMount.Commit.Repo.Name
+		commitID := commitMount.Commit.ID
+		var fromCommitID string
+		if commitMount.DiffMethod != nil && commitMount.DiffMethod.FromCommit != nil {
+			fromCommitID = commitMount.DiffMethod.FromCommit.ID
+		}
+		var fullFile bool
+		if commitMount.DiffMethod != nil {
+			fullFile = commitMount.DiffMethod.FullFile
+		}
+		shard := commitMount.Shard
+		fileInfos, err := c.ListFile(repo, commitID, "/", fromCommitID, fullFile, shard, false)
+		if err != nil {
+			return err
+		}
+		for _, fileInfo := range fileInfos {
+			if err := os.MkdirAll(filepath.Join(PFSInputPrefix, repo), 0777); err != nil {
+				return err
+			}
+			path := filepath.Join(PFSInputPrefix, repo, fileInfo.File.Path)
+			f, err := os.Create(path)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			w := bufio.NewWriter(f)
```

Comment:
Was there a reason you choose to use a buffered writer? Files normally have their own in memory buffering that's handled by the OS so we're probably better off just using that directly.